src/main.ts

```diff
@@ -0,0 +1,146 @@
+                return {
+                    message: `**PR submitted to Github** by ${account} with title: ${parsedSchema.pull_request.title}`,
+                    url: parsedSchema.pull_request.html_url,
+                    details: `Pr for merging ref ${parsedSchema.pull_request.head.ref} to base branch ${parsedSchema.pull_request.base.ref}`
+                };
+            case EventType.ISSUE:
+                parsedSchema = schemaOnIssue.validateSync(eventPayloadText);
+                account = parsedSchema.sender.login;
+                return {
+                    message: `**New/updated GitHub issue**: ${parsedSchema.issue.title}`,
+                    url: parsedSchema.issue.html_url,
+                    details: `Issue state: ${parsedSchema.issue.state}  - assignee: ${parsedSchema.issue.assignee}`
+                };
+            case EventType.ISSUE_COMMENT:
+                parsedSchema = schemaOnIssueComment.validateSync(eventPayloadText);
+                return {
+                    message: `**A Github issue comment was posted** by ${account} with content ${parsedSchema.comment.body}`,
+                    url: parsedSchema.issue.html_url,
+                    details: `Issue state: ${parsedSchema.issue.state}  - assignee: ${parsedSchema.issue.assignee}`
+                };
+            default:
+                return {
+                    message: eventName ? eventName.replace(/_/g, '.') : 'A GitHub Actions event has occurred',
+                    url: `https://github.com/${repository}/actions`,
+                    details: core.getInput('details')
+                };
+        }
+    } catch (error) {
+        let errorDetails;
+        if (error instanceof ValidationError) {
+            errorDetails = error.errors;
+        } else {
+            errorDetails = error.message;
+        }
+        return core.setFailed(`ERROR : ${errorDetails}`);
+    }
+};
+
+
+async function notifyTeams(notificationMessage: NotificationMessage) {
+    if (!(core.getInput('webhook_url'))) {
+        core.setFailed('Error : Webhook URL configuration is missing. Aborting');
+        process.exit(1);
+    }
+    const matches = core.getInput('webhook_url').match(/^https?:\/\/([^/?#]+)(.*)/i);
+    const hostnameMatch = matches && matches[1];
+    const pathMatch = matches && matches[2];
+
+    let requestBodyData = JSON.stringify(notificationTemplate);
+    requestBodyData = requestBodyData
+        .replace(/GITHUB_WORKFLOW/g, `${workflow}`)
+        .replace(/GITHUB_REPOSITORY/g, `${repository}`)
+        .replace(/GITHUB_REF/g, `${branch}`)
+        .replace(/GITHUB_TRIGGER_EVENT_DETAILS/g, `${notificationMessage.details}`)
+        .replace(/GITHUB_TRIGGER_EVENT/g, `${notificationMessage.message}`)
+        .replace(/GITHUB_EVENT_URL/g, `${notificationMessage.url}`)
+        .replace(/GITHUB_STATUS/g, `${core.getInput('job_status')}`);
+
+    const options = {
+        hostname: `${hostnameMatch}`,
```

Comment:
here and elsewhere in the code - why wrap variables in ticks?

```suggestion
        hostname: hostnameMatch,
```

Reply:
This has been removed with the changes for node-fetch.
In other places backticks are used for template literals, basically to allow string interpolation and concatenation.